### PR TITLE
Finish generator config file loading

### PR DESCRIPTION
**Pull request by winder**
## Summary

- Finish config file implementation. The parameters are now unmarshalled into an object and passed to the generator.
- Minor cleanup

## Test Plan

Made sure that `run_generator.sh` still works.

### CODE REVIEW

File: cmd/block-generator/generator/generate.go [hunk 1]
```diff
@@ -13,23 +14,62 @@ import (
...
+
+	// Asset configuration
+	AssetCreateFraction  float32 `mapstructure:"asset_create_fraction"`
+	AssetDestroyFraction float32 `mapstructure:"asset_destroy_fraction"`
+	AssetOptinFraction   float32 `mapstructure:"asset_optin_fraction"`
+	AssetCloseFraction   float32 `mapstructure:"asset_close_fraction"`
+	AssetXferFraction    float32 `mapstructure:"asset_xfer_fraction"`
+}
+
+func sumIsCloseToOne(numbers ...float32) bool {
```
Comment (brianolson) [inline cmd/block-generator/generator/generate.go:46]: Looks like most of the config fractions aren't used yet.
The other pattern I've seen is to just accept arbitrary float64 weights, sum them up, and then scale a random [0..1.0) to that sum and see where it lands in the lined up weights and pick that action.

Comment (winder) [reply cmd/block-generator/generator/generate.go:46]: Yup, that's next. I realized that this was a fairly complete piece of functionality, so decided to make the PR.

File: cmd/block-generator/generator/generate.go [hunk 2]
```diff
@@ -194,75 +238,87 @@ func (g *generator) initializeAccounting() {
...
+
+	return withAd
+}
+
+// generatePaymentTxn creates a new payment transaction. The sender is always a genesis account, the receiver is random,
+// or a new account.
 func (g *generator) generatePaymentTxn(round uint64, intra uint64) (types.SignedTxnWithAD, error) {
 	var receiveIndex uint64
-	if g.numPayments%g.config.NewAccountFrequency == 0 {
+	if g.numPayments%uint64(100*g.config.PaymentNewAccountFraction) == 0 {
```
[the remark this answers is not in the document]
Comment (winder) [reply cmd/block-generator/generator/generate.go:267]: It's not correct, I've fixed it in my followup PR

File: cmd/block-generator/generator/generate.go [hunk 3]
```diff
@@ -13,23 +14,62 @@ import (
...
-	GenesisHash                  sdk_types.Digest
+	Name                         string `mapstructure:"name"`
+	NumGenesisAccounts           uint64 `mapstructure:"genesis_accounts"`
+	GenesisAccountInitialBalance uint64 `mapstructure:"genesis_account_balance"`
+
+	// Block generation
+	TxnPerBlock uint64 `mapstructure:"tx_per_block"`
+
+	// TX Distribution
+	PaymentTransactionFraction float32 `mapstructure:"tx_pay_fraction"`
```
[the remark this answers is not in the document]
Comment (winder) [reply cmd/block-generator/generator/generate.go:31]: For the level of accuracy required, this is already more than enough.

File: cmd/block-generator/generator/generate.go [hunk 4]
```diff
@@ -13,23 +14,62 @@ import (
...
+	AssetCloseFraction   float32 `mapstructure:"asset_close_fraction"`
+	AssetXferFraction    float32 `mapstructure:"asset_xfer_fraction"`
+}
+
+func sumIsCloseToOne(numbers ...float32) bool {
+	var sum float32
+	for _, num := range numbers {
+		sum += num
+	}
+	return sum > 0.99 && sum < 1.01
```
[the remark this answers is not in the document]
Comment (winder) [reply cmd/block-generator/generator/generate.go:51]: Yes, I don't expect the error for floating-point numbers to be anywhere near this.

File: cmd/block-generator/generator/generate.go [hunk 5]
```diff
@@ -194,75 +238,87 @@ func (g *generator) initializeAccounting() {
...
-	*/
+	return convert(txn, types.ApplyData{}), nil
+}
 
-	withAd := types.SignedTxnWithAD{
-		SignedTxn: stxn,
-		ApplyData: types.ApplyData{},
-	}
-	return withAd, nil
+func (g *generator) generateAssetTxn(round uint64, intra uint64) (types.SignedTxnWithAD, error) {
```
[the remark this answers is not in the document]
Comment (winder) [reply cmd/block-generator/generator/generate.go:322]: Yes, I'm implementing this now in another branch

**Review by brianolson — APPROVED** [review]

**Pull request merged**